app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Successfully created new user #{@user.name}."
+      redirect_to users_path
+    else
+      flash.now[:error] = "Could not create new user."
+      render:new, status: :bad_request
+    end
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to user_path(@user)
```

Comment:
You've implemented all 7 RESTful controller actions for `User`, but I think you only need `index` and `show` for this site.